backend.py

```diff
@@ -60,6 +54,35 @@ def __repr__(self):
         return "<State> {} {}>".format(self.board, self.robots)
 
 
+class Rotation(Enum):
+    N = 0
+    E = 90
+    S = 180
+    W = 270
+
+    def __add__(self, other):
+        return Rotation(self.value + other.value)
+
+    def rotate(self, where_to):
+        if where_to == "right":
+            return Rotation((self.value + 90) % 360)
+        if where_to == "left":
+            return Rotation((self.value + 270) % 360)
+        if where_to == "upside_down":
+            return Rotation((self.value + 180) % 360)
+
+    def new_coordinates(self, coordinates):
+        (x, y) = coordinates
```

Comment:
the `x` and `y` are not needed here.

Reply:
Yes, thanks, the whole thing was rewritten.